### PR TITLE
Add unsubscribe page

### PR DESCRIPTION
**Pull request by MarkALeonard**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Introduces a new front controller for unsubscribing from the newsletter using an email address. Users can currently subscribe but cannot unsubscribe unless they have an account and login to change their preferences.
| Type?         |  new feature 
| BC breaks?    |  no
| Deprecations? |  no
| Fixed ticket? | Fixes PrestaShop/PrestaShop#21860
| How to test?  | Install module and navigate to [Domain]/module/ps_emailsubscription/unsubscription  unsubscribe

### CODE REVIEW

File: views/templates/front/unsubscription_execution.tpl
Comment (NeOMakinG) [inline views/templates/front/unsubscription_execution.tpl:20]:
```suggestion
{/block}

```

File: views/templates/front/unsubscription_execution.tpl
Comment (NeOMakinG) [inline views/templates/front/unsubscription_execution.tpl:18]:
```suggestion
  <div class="email_subscription block_newsletter" id="blockEmailSubscription_unsubscribe">
    {if $variables.msg}
      <p class="notification {if $variables.nw_error}notification-error{else}notification-success{/if}{$variables.msg}</p>
    {/if}

    <form action="{$urls.current_url}" method="post">
      <input type="text" name="email" value="{$variables.value}" placeholder="{l s='Your e-mail' d='Modules.Emailsubscription.Shop'}" />
      <p>{l s='Please enter your email to unsubscribe from our newsletter.' d='Modules.Emailsubscription.Shop'}</p>
      {hook h='displayNewsletterRegistration'}
      <input type="hidden" value="blockEmailSubscription_unsubscribe" name="blockHookName" />
      <input type="submit" value="ok" name="submitNewsletter" />
      <input type="hidden" name="action" value="1" />
    </form>
  </div>
```

Comment (NeOMakinG) [reply views/templates/front/unsubscription_execution.tpl:18]: Indentation isn't fine :)

File: views/templates/front/unsubscription_execution.tpl
Comment (NeOMakinG) [inline views/templates/front/unsubscription_execution.tpl:20]:
```suggestion
```

**Review by NeOMakinG — APPROVED** [review]

**Comment by Progi1984** [comment]
@PrestaShop/product-team Ok for adding this unsubscribe feature to the module ?

**Comment by kpodemski** [comment]
Do we have any confirmation email for unsubscribe action? Or we can type any email we want?

**Comment by LouiseBonnard** [comment]
Related to https://github.com/PrestaShop/PrestaShop/issues/21860.

**Comment by kpodemski** [comment]
I think we would need something more complex, not just possibility to unsubscribe every email address we want, IMO:

- we need a way to unsubscribe via URL with proper hash
- 2 step unsubscribe by request, first request via form, then confirmation link send to email (with proper hash)

what y'all think @PrestaShop/product-team ?

**Comment by matks** [comment]
No answer yet from product team, I send a kind reminder with hugs in https://github.com/PrestaShop/PrestaShop/issues/21860 ❤️

**Comment by marionf** [comment]
Great idea @MarkALeonard, I don't know if it should be done in this PR or in another one but we should also add an Unsubscribe link in all emails.
This PR should be checked by @PrestaShop/ux-team before being merged because the current design does not seem optimal 

Capture d’écran 2021-07-30 à 11 57 14

**Comment by prestascott** [comment]
Hello everyone,

Thanks @MarkALeonard for your request!

Users need to be able to unsubscribe from the newsletter whenever they want. 

In fact, I'll replace the label of the button with "Unsubscribe" instead of "ok" because the action button label needs to be explicit for users @marionf. Am I right? @Julievrz 

Also, I would erase the text helper at the bottom of the input and place it at the top so it's more clear.

This is my design proposition:
unsubscribe

Wdyt?

File: controllers/front/unsubscription.php
Comment (PierreRambaud) [inline controllers/front/unsubscription.php:55]: Method `ajaxDie` is deprecated, use `ajaxRender` instead

File: controllers/front/unsubscription.php
Comment (PierreRambaud) [inline controllers/front/unsubscription.php:51]: Shouldn't do something if the form is not valid, and had no error?

File: views/templates/front/unsubscription_execution.tpl
Comment (PierreRambaud) [inline views/templates/front/unsubscription_execution.tpl:12]: make sure value is escaped

File: views/templates/front/unsubscription_execution.tpl
Comment (PierreRambaud) [inline views/templates/front/unsubscription_execution.tpl:8]: Same about msg var, make sure it's escaped

**Comment by xbkappa** [comment]
having problems to show the form in the front office, how can i show it?

**Comment by matks** [comment]
Hello, it seems this PR is stale since August.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity".

If however you are still interested in it, please ping us 😉  we can reopen.

Thank you for your contribution.

**Comment by giuggiolafra** [comment]
Hello, it's possibile to have an update of this module?

**Pull request closed without merging**
